tests/ludwig/config_validation/test_checks.py

```diff
@@ -198,3 +198,20 @@ def test_dense_binary_encoder_0_layer():
     }
     with pytest.raises(ConfigValidationError):
         ModelConfig.from_dict(config)
+
+
+def test_onehot_category_encoder():
+    config = {
+        "defaults": {"category": {"encoder": {"type": "onehot"}}},
+        "input_features": [
+            {"name": "MSSubClass", "type": "category"},
+            {"name": "MSZoning", "type": "category"},
+            {"name": "Street", "type": "category"},
+            {"name": "Neighborhood", "type": "category"},
+        ],
+        "model_type": "ecd",
+        "output_features": [{"name": "SalePrice", "type": "number"}],
+        "trainer": {"train_steps": 1},
+        "combiner": {"type": "concat", "num_fc_layers": 2},
+    }
+    ModelConfig.from_dict(config)
```

Comment:
@abidwael does just doing `.from_dict()` repro the error? I would image this would require at least 1-2 steps of training for the error in your PR comment to be reproed

Reply:
Agreed, it doesn't. Will remove this test as it will be captured in https://github.com/ludwig-ai/ludwig/pull/2991

Reply:
I would probably add a test here as well since these will run on every commit, whereas the tests in #2991 will run on merge to master? or is that not the case?

Reply:
It will run on every commit